presto-hive/src/main/java/io/prestosql/plugin/hive/s3/HiveS3Config.java

```diff
@@ -109,6 +113,28 @@ public HiveS3Config setS3SignerType(PrestoS3SignerType s3SignerType)
         return this;
     }
 
+    public String getS3SignerClass()
+    {
+        return s3SignerClass;
+
+    }
+
+    @Config("hive.s3.signer-class")
+    public HiveS3Config setS3SignerClass(String s3SignerClass)
+    {
+        this.s3SignerClass = s3SignerClass;
+        Class<? extends Signer> klass;
```

Comment:
Hi!! Can we move this logic of loading the class and registering, outside of `HiveS3Config` ?

Reply:
Better place for that would be `io.prestosql.plugin.hive.s3.PrestoS3ConfigurationInitializer`

Reply:
I've moved the logic to PresoS3FileSystem. The custom signer class is per catalog now and not per connector.